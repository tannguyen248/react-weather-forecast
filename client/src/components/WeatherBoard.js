import React from "react";
import styled from "styled-components";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Card from "react-bootstrap/Card";
import WeatherCard from "./WeatherCard";
import OtherLocations from "./OtherLocations";

const WeatherBoard = ({ location, locations, handleClickChangeLocation }) => {
  if (!locations || !location) {
    return null;
  }

  return (
    <Card data-testid="card-weather-board">
      <Card.Header>
        <OtherLocations
          currentId={location.woeid}
          locations={locations}
          handleClick={handleClickChangeLocation}
        />
      </Card.Header>
      <Card.Body>
        <Row>
          {location.weathers.map((state, index) => (
            <Col key={index} xs={6} md={4} lg="auto">
              <Content>
                <WeatherCard {...state} />
              </Content>
            </Col>
          ))}
        </Row>
      </Card.Body>
    </Card>
  );
};

const Content = styled.div`
  margin-top: 5px;
`;

export default WeatherBoard;
