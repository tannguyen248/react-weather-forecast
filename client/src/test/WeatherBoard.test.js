import renderer from "react-test-renderer";
import WeatherBoard from "../components/WeatherBoard";

describe("WeatherBoard", () => {
  it("should not display weather board when data wrong", () => {
    const { location, locations } = { location: null, locations: null };
    const tree = renderer
      .create(<WeatherBoard {...{ location, locations }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should display weather board correctly", () => {
    const locations = [
      {
        title: "San Francisco",
        location_type: "City",
        woeid: 2487956,
        latt_long: "37.777119, -122.41964",
      },
      {
        title: "San Diego",
        location_type: "City",
        woeid: 2487889,
        latt_long: "32.715691,-117.161720",
      },
    ];

    const location = {
      title: "Title",
      woeid: "44418",
      weathers: [
        {
          title: 'Title',
          state: 'Snow',
          stateAbbr: 's',
          minTemp: 5,
          maxTemp: 10,
          date: new Date("2020-01-01"),
          dayOfWeek: 2,
        },
      ],
    };
    const tree = renderer
      .create(<WeatherBoard {...{ location, locations }} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
