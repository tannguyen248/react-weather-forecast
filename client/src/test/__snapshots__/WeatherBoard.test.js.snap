// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherBoard should display weather board correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    <div
      className="nav card-header-pills nav-pills"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="2487956"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          San Francisco
        </a>
      </div>
      <div
        className="nav-item"
        onClick={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key="2487889"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          San Diego
        </a>
      </div>
    </div>
  </div>
  <div
    className="card-body"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-auto col-md-4 col-6"
      >
        <div
          className="sc-gsTCUz gAVgso"
        >
          <div
            className="text-center card"
          >
            <img
              className="card-img-top"
              src="https://www.metaweather.com/static/img/weather/s.svg"
            />
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
              >
                2
                 (
                <span>
                  1
                  -
                  1
                </span>
                )
              </div>
              <p
                className="card-text"
              >
                <span>
                  10
                  <sup
                    className="sc-bdfBwQ kqJIKH"
                  >
                    o
                  </sup>
                </span>
                 -
                 
                <span>
                  5
                  <sup
                    className="sc-bdfBwQ kqJIKH"
                  >
                    o
                  </sup>
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeatherBoard should not display weather board when data wrong 1`] = `null`;
